### PR TITLE
feat: add client upload time

### PR DESCRIPTION
**Pull request by Mercy811**
### Summary

https://amplitude.atlassian.net/browse/AMP-87178

### Checklist

* [ ] Does your PR title have the correct [title format](https://github.com/amplitude/Amplitude-TypeScript/blob/main/CONTRIBUTING.md#pr-commit-title-conventions)?
* Does your PR have a breaking change?:

### CODE REVIEW

File: packages/analytics-types/src/payload.ts [hunk 1]
```diff
@@ -8,4 +8,5 @@ export interface Payload {
   api_key: string;
   events: readonly Event[];
   options?: PayloadOptions;
+  client_upload_time?: string;
```
Comment (kevinpagtakhan) [inline packages/analytics-types/src/payload.ts:11]: Is this different from `time`?

Comment (Mercy811) [reply packages/analytics-types/src/payload.ts:11]: yes, they are different. `time` in SDK is [`client_event_time` in background](https://github.com/amplitude/amplitude/blob/90b19342d7d991bb4854102ad85fda22b299a3b3/event/converters/http_event.py#L254-L258).

Comment (liuyang1520) [reply packages/analytics-types/src/payload.ts:11]: Same question, seems similar as `time` field except that this is ISO format.

Comment (Mercy811) [reply packages/analytics-types/src/payload.ts:11]: Please see more background [here](https://amplitude.slack.com/archives/C02E1H87E2C/p1697495952472679)

**Review by liuyang1520 — APPROVED** [review]
Discussed with @Mercy811 offline, `client_upload_time` can be used by the server to estimate the clock drift of that device (server received time - upload time), then adjust the client event time in the server side.

File: packages/analytics-types/src/payload.ts (hunk 1 above)
Comment (Mercy811) [reply packages/analytics-types/src/payload.ts:11]: Closed it as getting approved from Kevin from mid week oncall sync

Comment (ondrej) [inline packages/analytics-types/src/payload.ts:11]: Are there cases where we wouldn't set `client_upload_time`?

**Pull request merged**